components/expression/condition.jsx

```diff
@@ -0,0 +1,238 @@
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/*
+ * Callbacks for various expression condition events such as value change, delete etc
+ */
+	events: PropTypes.shape({
+		onChangeResource: PropTypes.func,
+		onChangeOperator: PropTypes.func,
+		onChangeValue: PropTypes.func,
+		onDelete: PropTypes.func,
+	}).isRequired,
+	/**
+	 * **Text labels for internationalization**
+	 * This object is merged with the default props object on every
+	 * * `deleteCondition`: Title for the delete condition button. Defaults to "Delete Condition".
+	 * * `label`: Label for the condition, shown left-most in the row. Left empty on default.
+	 * * `operator`: Label for the operator selection dropdown. Defaults to "Operator"
+	 * * `resource`: Label for the resource selection dropdown. Defaults to "Resource"
+	 * * `value`: Label for the value input box. Defaults to "Value"
+	 */
+	labels: PropTypes.shape({
+		deleteCondition: PropTypes.string,
+		label: PropTypes.string,
+		operator: PropTypes.string,
+		resource: PropTypes.string,
+		value: PropTypes.string,
+	}),
+	/**
+	 * Controls whether the condition is a sub-condition inside a ExpressionGroup
+	 */
+	isSubCondition: PropTypes.bool,
+	/**
+	 * **Array of item objects that are options in the resource selection dropdown menu.**
+	 * Each object can contain:
+	 * * `id`: A unique identifier string.
+	 * * `label`: A primary string of text for a menu item.
+	 * ```
+	 * {
+	 * 	id: '1',
+	 * 	label: 'Resource 1',
+	 * },
+	 * ```
+	 * Note: The dropdown uses the Combobox Component, and `resourcesList` is
+	 * passed as `options` props to it, and hence shall also support more
+	 * custom objects. Please refer to the Combobox documentation.
+	 */
+	resourcesList: PropTypes.arrayOf(PropTypes.object),
+	/**
+	 *  Accepts an object from the `resourcesList` which needs to be selected
+	 *  for the resource dropdown menu,
+	 */
+	resourceSelected: PropTypes.object,
+	/**
+	 * **Array of item objects that are options in the operator selection dropdown menu.**
+	 * Each object can contain:
+	 * * `id`: A unique identifier string.
+	 * * `label`: A primary string of text for a menu item.
+	 * ```
+	 * {
+	 * 	id: '1',
+	 * 	label: 'Operator 1',
+	 * },
+	 * ```
+	 * Note: The dropdown uses the Combobox Component, and `operatorList` is
+	 * passed as `options` props to it, and hence shall also support more
+	 * custom objects. Please refer to the Combobox documentation.
+	 */
+	operatorsList: PropTypes.arrayOf(PropTypes.object),
+	/**
+	 *  Accepts an object from the `operatorSelected` which needs to be selected
+	 *  for the operator dropdown menu,
+	 */
+	operatorSelected: PropTypes.object,
+	/**
+	 *   Sets the input value for the value input.
+	 */
+	value: PropTypes.string,
+};
+
+const defaultProps = {
+	assistiveText: {
+		title: 'Condition',
+		deleteIcon: 'Delete Condition',
+	},
+	isSubCondition: false,
```

Comment:
This default is not needed, since if it's not passed in it will be `undefined` and equate to falsey values in conditionals